### PR TITLE
Allow admins/editors to fetch users with Cognito Identity Id

### PR DESCRIPTION
**Pull request by gshaw1997**
**Purpose of this PR**
Adds route to fetch users by id. Additionally, adds logic to allow admins/editors the ability to get additional user properties, in this case the user's Cognito Identity Id.
Completes https://app.clubhouse.io/clarkcan/story/880/point-s3-driver-in-file-manager-to-new-s3-bucket

**Changes in this PR**
- Removed `@cyber4all/clark-entity` package, swapped for internal interfaces (Moved to https://github.com/Cyber4All/user-service/pull/151)
- Add optional `cognitoIdentityId` property to the user interface (Moved to https://github.com/Cyber4All/user-service/pull/151)
- Moved typings to shared (Moved to https://github.com/Cyber4All/user-service/pull/151)
- Move `AuthorizationManager` to shared (Moved to https://github.com/Cyber4All/user-service/pull/152)
- Swap internal `ServiceModule` for `node-service-module` pacakge (Moved to https://github.com/Cyber4All/user-service/pull/153)
- Update parameters for `getOpenIdToken` to accept `username` and `isAdminOrEditor`
- Expose route to fetch user data

### CODE REVIEW

**Comment by gshaw1997** [comment]
> This PR seems to of snowballed a little bit, is it possible to move the entity changes and file relocations to their own PR?

In order to add the `cognitoIdentityId` property, the entity package needed to be uninstalled and move to an internal type. Since that package is no longer maintained it would have been more cumbersome to try to update the package

File: src/UserMetaRetriever/Interactor.ts [hunk 1]
```diff
@@ -11,6 +12,38 @@ namespace Drivers {
   export const datastore = () => Module.resolveDependency(UserMetaDatastore);
 }
 
+namespace Gateways {
```
Comment (jbarlow96) [inline src/UserMetaRetriever/Interactor.ts]: Just for clarification, can you explain why you used namespace here?

**Comment by DonSeannelly** [comment]
> > This PR seems to of snowballed a little bit, is it possible to move the entity changes and file relocations to their own PR?
> 
> @DonSeannelly In order to add the `cognitoIdentityId` property, the entity package needed to be uninstalled and move to an internal type. Since that package is no longer maintained it would have been more cumbersome to try to update the package

I think what I was hoping for in an ideal world is to have incremental PRs when there are large library or internal directory changes, where the first PRs would be pure refactorings without any changes or enhancements. They could then be reviewed and deployed in isolation, and the final PR with logical changes to the service would be more concise and easy to review.

I think this is coming more from experiences with Learning Object Service, where several devs would be woking on it at the same time, and so the higher delta between branches, the more headaches we had.

I'm good to leave this as is for now, but I think I found my first guideline to write up as Service Owner here.

**Comment by gshaw1997** [comment]
> > > This PR seems to of snowballed a little bit, is it possible to move the entity changes and file relocations to their own PR?
> > 
> > 
> > @DonSeannelly In order to add the `cognitoIdentityId` property, the entity package needed to be uninstalled and move to an internal type. Since that package is no longer maintained it would have been more cumbersome to try to update the package
> 
> I think what I was hoping for in an ideal world is to have incremental PRs when there are large library or internal directory changes, where the first PRs would be pure refactorings without any changes or enhancements. They could then be reviewed and deployed in isolation, and the final PR with logical changes to the service would be more concise and easy to review.
> 
> I think this is coming more from experiences with Learning Object Service, where several devs would be woking on it at the same time, and so the higher delta between branches, the more headaches we had.
> 
> I'm good to leave this as is for now, but I think I found my first guideline to write up as Service Owner here.

Got it, that makes sense. I'll break refactorings out into separate PRs

File: src/UserMetaRetriever/Interactor.ts (hunk 1 above)
Comment (gshaw1997) [reply src/UserMetaRetriever/Interactor.ts]: @jbarlow96 The `namespace` is used here to organize and group similar things, in this case types of dependencies.

**Pull request merged**